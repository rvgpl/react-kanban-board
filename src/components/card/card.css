.card {
  background: white;
  border-radius: 4px;
  padding: 1rem;
}

.card-text {
  font-size: 1rem;
  line-height: 1.3;
  margin-bottom: 1.5rem;
}

.card-footer {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.card-timestamp {
  font-size: 1rem;

}