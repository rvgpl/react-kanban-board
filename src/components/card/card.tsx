import React from "react";
import { ITask } from "../../types";
import { formatDate } from "../../utils";
import { Button } from "../button/button";

import "./card.css";

type ICardProps = {
  onClick: (cardId: string | number) => void;
  disableButton?: boolean;
  task: ITask;
};

const Card: React.FC<ICardProps> = ({
  onClick,
  task,
  disableButton = false,
}) => {
  return (
    <div className="card">
      <p className="card-text">{task.text}</p>
      <div className="card-footer">
        <time className="card-timestamp">{formatDate(task.createdAt)}</time>
        {!disableButton && (
          <Button onClick={() => onClick(task.id)}>Move &gt;</Button>
        )}
      </div>
    </div>
  );
};

export { Card };
