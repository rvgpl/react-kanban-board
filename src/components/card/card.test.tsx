import { render, fireEvent } from "@testing-library/react";
import { Card } from "./card";

const sampleTask = {
  id: 1,
  text: "Task 1",
  createdAt: 1646412499232,
};
describe("Card", () => {
  it("should render", () => {
    const { container } = render(
      <Card onClick={jest.fn()} task={sampleTask} />
    );
    expect(container).toMatchSnapshot();
  });
  it("should not render button on disableButton is true", () => {
    const { container } = render(
      <Card onClick={jest.fn()} task={sampleTask} disableButton />
    );
    expect(container).toMatchSnapshot();
  });
  it("should call onClick prop on click of button", () => {
    const onClickMock = jest.fn();
    const { getByText } = render(
      <Card onClick={onClickMock} task={sampleTask} />
    );
    fireEvent.click(getByText("Move >"));
    expect(onClickMock).toBeCalledTimes(1);
  });
});
