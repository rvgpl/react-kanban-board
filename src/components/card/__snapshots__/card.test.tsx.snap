// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should not render button on disableButton is true 1`] = `
<div>
  <div
    class="card"
  >
    <p
      class="card-text"
    >
      Task 1
    </p>
    <div
      class="card-footer"
    >
      <time
        class="card-timestamp"
      >
        4-3-2022 20:48
      </time>
    </div>
  </div>
</div>
`;

exports[`Card should render 1`] = `
<div>
  <div
    class="card"
  >
    <p
      class="card-text"
    >
      Task 1
    </p>
    <div
      class="card-footer"
    >
      <time
        class="card-timestamp"
      >
        4-3-2022 20:48
      </time>
      <button
        class="button"
        data-testid="button"
      >
        Move &gt;
      </button>
    </div>
  </div>
</div>
`;
